nutils/evaluable.py

```diff
@@ -2340,27 +2361,32 @@ def evalf(self, index):
 
 class Eig(Evaluable):
 
-  __slots__ = 'symmetric', 'func'
+  __slots__ = 'symmetric', 'func', 'w_dtype', 'vt_dtype'
 
   @types.apply_annotations
   def __init__(self, func:asarray, symmetric:bool=False):
     assert func.ndim >= 2 and func.shape[-1] == func.shape[-2]
     self.symmetric = symmetric
     self.func = func
+    self.w_dtype = float if symmetric else complex
+    self.vt_dtype = func.dtype if symmetric else complex
```

Comment:
I think `self.vt_dtype` should at least be a float.

Reply:
done, thanks.